// prismjs-terminal/dist/commonjs/themes/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.plain = exports.prismDark = exports.github = exports.moria = exports.debug = exports.xonokai = void 0;
/**
 * Port of the `xonokai` PrismJS theme.
 */
var xonokai_js_1 = require("./xonokai.js");
Object.defineProperty(exports, "xonokai", { enumerable: true, get: function () { return xonokai_js_1.theme; } });
/**
 * This theme just shows which token names are in use The list of tokens is
 * everything defined in the tsx PrismJS language grammar, since that is a
 * superset of JavaScript, TS, JSX, and HTML.
 */
var debug_js_1 = require("./debug.js");
Object.defineProperty(exports, "debug", { enumerable: true, get: function () { return debug_js_1.theme; } });
/**
 * Inspired by vim Moria theme
 * https://github.com/vim-scripts/moria
 *
 * It's not *quite* as faithful as I'd like (and I *would* like, since it's the
 * color scheme I personally use in Vim, and so most natural for me), because
 * Vim and PrismJS make slightly different choices about how they label program
 * tokens.
 */
var moria_js_1 = require("./moria.js");
Object.defineProperty(exports, "moria", { enumerable: true, get: function () { return moria_js_1.theme; } });
/**
 * Port of GHColors prismjs theme.
 * <https://github.com/PrismJS/prism-themes/blob/057c2a4430d78268528e65ba92860703c9cb56d8/themes/prism-ghcolors.css>
 */
var github_js_1 = require("./github.js");
Object.defineProperty(exports, "github", { enumerable: true, get: function () { return github_js_1.theme; } });
/**
 * Port of the `prism-dark` PrismJS theme.
 * <https://github.com/PrismJS/prism/blob/867804573562ef0ac5acf470420b97e43488ccde/themes/prism-dark.css>
 *
 * Slightly tweaked a few colors so that they are a bit more readable on a
 * terminal, some terminals have rendering issues when contrast is too low.
 */
var prism_dark_js_1 = require("./prism-dark.js");
Object.defineProperty(exports, "prismDark", { enumerable: true, get: function () { return prism_dark_js_1.theme; } });
/**
 * No styling, just the code as plain text.
 * Will add line numbers and padding if requested, of course.
 */
var plain_js_1 = require("./plain.js");
Object.defineProperty(exports, "plain", { enumerable: true, get: function () { return plain_js_1.theme; } });
